x-pack/plugins/case/server/client/configure/utils.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ConnectorField,
+  ConnectorMappingsAttributes,
+  ConnectorTypes,
+} from '../../../common/api/connectors';
+import {
+  JiraGetFieldsResponse,
+  ResilientGetFieldsResponse,
+  ServiceNowGetFieldsResponse,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../actions/server/types';
+
+const normalizeJiraFields = (jiraFields: JiraGetFieldsResponse): ConnectorField[] =>
+  Object.keys(jiraFields).reduce<ConnectorField[]>(
+    (acc, data) =>
+      jiraFields[data].schema.type === 'string'
+        ? [
+            ...acc,
+            {
+              id: data,
+              name: jiraFields[data].name,
+              required: jiraFields[data].required,
+              type: 'text',
+            },
+          ]
+        : acc,
+    []
+  );
+
+const normalizeResilientFields = (resilientFields: ResilientGetFieldsResponse): ConnectorField[] =>
+  resilientFields.reduce<ConnectorField[]>(
+    (acc: ConnectorField[], data) =>
+      (data.input_type === 'textarea' || data.input_type === 'text') && !data.read_only
+        ? [
+            ...acc,
+            {
+              id: data.name,
+              name: data.text,
+              required: data.required === 'always',
+              type: data.input_type,
+            },
+          ]
+        : acc,
+    []
+  );
+const normalizeServiceNowFields = (snFields: ServiceNowGetFieldsResponse): ConnectorField[] =>
+  snFields.reduce<ConnectorField[]>(
+    (acc, data) => [
+      ...acc,
+      {
+        id: data.element,
+        name: data.column_label,
+        required: data.mandatory === 'true',
+        type: parseFloat(data.max_length) > 160 ? 'textarea' : 'text',
```

Comment:
Is `max_length` always presented?

Reply:
yes

Reply:
look in `x-pack/plugins/actions/server/builtin_action_types/servicenow/service.ts`

the definition of `fieldsUrl` explicitly tells the api what fields to return: `&sysparm_fields=max_length,element,column_label,mandatory`